omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/VisitDatesFragmentController.java

```diff
@@ -37,9 +39,12 @@ public SimpleObject setDuration(@SpringBean("visitService") VisitService visitSe
                                             @RequestParam(value="stopDate", required = false) Date stopDate,
                                             HttpServletRequest request, UiUtils ui) {
 
+        Boolean allowChangingVisitTime = new Boolean(Context.getAdministrationService().getGlobalProperty(CoreAppsConstants.GP_ALLOW_CHANGING_VISIT_TIME));
 
-        if (!isSameDay(startDate, visit.getStartDatetime())) {
+        if (!allowChangingVisitTime && !isSameDay(startDate, visit.getStartDatetime())) {
             visit.setStartDatetime(new DateTime(startDate).toDateMidnight().toDate());
+        }else{
+            visit.setStartDatetime(new DateTime(startDate).toDate());
```

Comment:
Do you still want this to be executed if `allowChangingVisitTime = false` but `isSameDay(startDate, visit.getStartDatetime()) == true`?

Reply:
I'll let @icrc-toliveira  answer but I do think this is the correct behavior... we only want to "reset" the time component to midnight if allow change visit time is false and we are changing the date... if allowChangeVisitTime is true, we always want to set the visit time to the time passed in via the submit.

Reply:
Hi  @dkayiwa and @mogoodrich 

I just did a small update, this should account for @dkayiwa concerns, while keeping the behavior " reset the time to midnight if allow change visit time is false and we are changing the date"

Reply:
@icrc-toliveira i find that this logic would be easier to understand if you put the `allowChangingVisitTime` below `if (!isSameDay(startDate, visit.getStartDatetime())) {`

That way, you do not have to repeat `!isSameDay(startDate, visit.getStartDatetime())`

Reply:
Done